src/rust/jsg-test/lib.rs

```diff
@@ -0,0 +1,200 @@
+use jsg::v8;
+use kj_rs::KjOwn;
+
+#[cxx::bridge(namespace = "workerd::rust::jsg_test")]
+mod ffi {
+    #[namespace = "workerd::rust::jsg"]
+    unsafe extern "C++" {
+        include!("workerd/rust/jsg/ffi.h");
+
+        type Isolate = jsg::v8::ffi::Isolate;
+    }
+
+    unsafe extern "C++" {
+        include!("workerd/rust/jsg-test/ffi.h");
+
+        type TestHarness;
+
+        pub unsafe fn create_test_harness() -> KjOwn<TestHarness>;
+        pub unsafe fn run_in_context(self: &TestHarness, callback: unsafe fn(*mut Isolate));
+    }
+}
+
+pub struct Harness(KjOwn<ffi::TestHarness>);
+
+impl Harness {
+    pub fn new() -> Self {
+        Self(unsafe { ffi::create_test_harness() })
+    }
+
+    pub fn run_in_context(&self, callback: fn(*mut v8::ffi::Isolate)) {
+        unsafe { self.0.run_in_context(callback) }
+    }
+}
+
+impl Default for Harness {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use jsg::Lock;
+    use jsg::Struct;
+    use jsg::v8;
+    use jsg::v8::ToLocalValue;
+    use jsg_macros::jsg_struct;
+
+    #[jsg_struct]
+    struct TestStruct {
+        pub str: String,
+    }
+
+    #[jsg_struct]
+    struct MultiPropertyStruct {
+        pub name: String,
+        pub age: u32,
+        pub active: String,
+    }
+
+    #[jsg_struct]
+    struct NestedStruct {
+        pub inner: String,
+    }
```

Comment:
Can we not test resource call bindgen? If not, perhaps we need macro expansion output based tests here? Would help to work through arg and ret value edge cases. I really hope we can flesh out more of these basics for function bindgen before rolling out too much code coverage on this.